src/hooks/useResource.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import axios from 'axios';
+import { useDispatch, useSelector } from 'react-redux';
+import { recordsSlice } from '../redux/slices';
+
+const useRecords = (resource) => {
+    const dispatch = useDispatch();
+    const records = useSelector((state) => state.records);
+
+    useEffect(() => {
+        (async (resource) => {
```

Comment:
A more readable pattern is to create a named async function (you can define it elsewhere for better testability and call it inside the effect (rather than invoking an anonymous function.